src/js/other-websites/icon.js

```diff
@@ -0,0 +1,452 @@
+  const existingUnderlay = document.getElementById("relay-popover-underlay");
+  const underlay = existingUnderlay ?? document.createElement("div");
+  underlay.id = "relay-popover-underlay";
+  underlay.style.display = "initial";
+  underlay.style.position = "fixed";
+  underlay.style.inset = "0";
+  underlay.style.zIndex = "99999999";
+  underlay.style.backgroundColor = "rgba(0, 0, 0, 0.1)";
+  const existingIframe = underlay.querySelector("iframe");
+  const iframe = existingIframe ?? document.createElement("iframe");
+  iframe.src = browser.runtime.getURL("inpage-panel.html");
+  iframe.style.position = "absolute";
+  iframe.style.width = `${RELAY_INPAGE_MENU_WIDTH}px`;
+  iframe.style.maxWidth = `95vw`;
+  positionIframe(iframe, target);
+  const onResizeWindow = () => {
+    positionIframe(iframe, target);
+  };
+  window.addEventListener("resize", onResizeWindow);
+  const onScroll = () => {
+    positionIframe(iframe, target);
+  };
+  window.addEventListener("scroll", onScroll);
+
+  /**
+   * @param {unknown} message 
+   */
+  const onInsertMask = (message) => {
+    if (
+      typeof message === "object" &&
+      typeof message?.filter === "string" &&
+      message.filter === "fillInputWithAlias" &&
+      typeof message.newRelayAddressResponse === "object" &&
+      (
+        typeof message.newRelayAddressResponse?.full_address === "string" ||
+        typeof message.newRelayAddressResponse?.address === "string"
+      )
+    ) {
+      // When generating a new mask, `newRelayAddressResponse` contains a mask object
+      // with the `full_address` property. When selecting an existing mask,
+      // however, it only has the properties `address` and `currentDomain`, but
+      // `address` does contain the full address (i.e. including @mozmail.com):
+      target.value = message.newRelayAddressResponse.full_address ?? message.newRelayAddressResponse.address;
+      target.dispatchEvent(new Event("input", { bubbles: true }));
+      closeMenu();
+    }
+  };
+  browser.runtime.onMessage.addListener(onInsertMask);
+
+  /**
+   * There's still no good way to get browsers to resize iframes to their
+   * contents — packages like iframe-resizer are stil in use… However, we just
+   * have the page in the iframe report its height to use whenever it resizes,
+   * and update accordingly:
+   * @param {unknown} message 
+   */
+  const onUpdateIframeHeight = (message) => {
+    if (
+      typeof message === "object" &&
+      typeof message?.method === "string" &&
+      message.method === "updateIframeHeight" &&
+      typeof message?.height !== "undefined"
+    ) {
+      iframe.style.height = `${message.height}px`;
+      positionIframe(iframe, target);
+    }
+  };
+  browser.runtime.onMessage.addListener(onUpdateIframeHeight);
+
+  /**
+   * @param {MouseEvent} _event 
+   */
+  const onUnderlayClick = (_event) => {
+    closeMenu();
+  };
+  underlay.addEventListener("click", onUnderlayClick);
```

Comment:
Question/thought: Do we need any [`touchstart`](https://developer.mozilla.org/en-US/docs/Web/API/Element/touchstart_event) support for the Android version of this add-on?

Reply:
I don't _think_ so, as [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event#safari_mobile) mentions a bug in old Safari versions for `click` events, implying that they're supposed to fire, and [StackOverflow says so](https://stackoverflow.com/a/37273344). But I'm not sure how to verify - do you know if there's an easy way to test `.xpi` files on Firefox mobile?